Core/include/Acts/Material/GridSurfaceMaterial.hpp

```diff
@@ -0,0 +1,194 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Material/ISurfaceMaterial.hpp"
+#include "Acts/Material/MaterialSlab.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+#include "Acts/Utilities/GridAccessHelpers.hpp"
+#include "Acts/Utilities/VectorHelpers.hpp"
+
+#include <vector>
+
+namespace Acts {
+
+/// @brief  This is an accessor for cases where the material is directly stored
+/// in the grid, it simply forwards the grid entry in const and non-const way.
+struct GridMaterialAccessor {
+  /// @brief  Direct const access to the material slap sorted in the grid
+  /// @tparam grid_type the type of the grid, also defines the point type
+  /// @param grid the grid
+  /// @param point the lookup point (already casted from global, or filled from local)
+  ///
+  /// @return the material slab from the grid bin associated to the lookup point
+  template <typename grid_type>
+  inline MaterialSlab& slab(grid_type& grid,
+                            const typename grid_type::point_t& point) const {
+    return grid.atPosition(point);
+  }
+
+  /// @brief Scale the material (by scaling the thickness)
+  ///
+  /// @param grid the grid (ignored)
+  /// @param scale the amount of the scaling
+  ///
+  /// @note this is not particularly fast
+  template <typename grid_type>
+  void scale(grid_type& grid, ActsScalar scale) {
+    for (auto& m : grid.values()) {
+      m.scaleThickness(scale);
+    }
+  }
+};
+
+/// @brief  This is an accessor for cases where the material is filled in a vector
+/// and then indexed by the grid
+struct IndexedMaterialAccessor {
+  /// @brief The internal storage of the material
+  std::vector<MaterialSlab> material;
+  /// @brief  Direct const access to the material slap sorted in the grid
+  /// @tparam grid_type the type of the grid, also defines the point type
+  /// @param grid the grid
+  /// @param point the lookup point (already casted from global, or filled from local)
+  ///
+  /// @return the material slab from the grid bin associated to the lookup point
+  template <typename grid_type>
+  inline const MaterialSlab& slab(
+      const grid_type& grid, const typename grid_type::point_t& point) const {
+    auto index = grid.atPosition(point);
+    return material[index];
+  }
+
+  /// @brief Scale the material (by scaling the thickness)
+  ///
+  /// @param grid the grid (ignored)
+  /// @param scale the amount of the scaling
+  template <typename grid_type>
+  void scale([[maybe_unused]] grid_type& grid, ActsScalar scale) {
```

Comment:
I would rather
```suggestion
  void scale(grid_type& /*grid*/, ActsScalar scale) {
```

Reply:
Done, had to update documentation as well though.